### ISSUE TITLE
Remote disk loses power, then recovers. Spyder not satisfied.

### ISSUE THREAD

**Issue by sound-fx** [spyder-ide/spyder#5329]
## Description

**What steps will reproduce the problem?**

1. I lost power to the remote machine hosting the drive with most of the *.py loaded in the Spyder editor, causing that machine to reboot.
2. After that machine came back online, I tried to save my recent changes in Spyder's editor.
3. This error message appeared in response to that failed attempt.

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

  File "/home/ionoadmin/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py", line 912, in set_working_directory
    refresh_console=False)
  File "/home/ionoadmin/anaconda2/lib/python2.7/site-packages/spyder/plugins/workingdirectory.py", line 286, in chdir
    os.chdir(directory)
OSError: [Errno 13] Permission denied: '/mnt/puff/MASE/HiCIRF-lite-AFCAP/Python/Lib/afcap/tests'

## Version and main components

* Spyder Version: 3.2.3 
* Python Version: 2.7.13
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.5.0  :  1.6.0 (OK)
pycodestyle >=2.3 :  2.3.1 (OK)
pygments >=2.0    :  2.2.0 (OK)
pandas >=0.13.1   :  0.20.3 (OK)
numpy >=1.7       :  1.13.1 (OK)
sphinx >=0.6.6    :  1.6.3 (OK)
rope >=0.9.4      :  0.9.4 (OK)
jedi >=0.9.0      :  0.9.0 (OK)
psutil >=0.3      :  5.2.0 (OK)
nbconvert >=4.0   :  5.1.1 (OK)
sympy >=0.7.3     :  1.1.1 (OK)
cython >=0.21     :  0.26 (OK)
qtconsole >=4.2.0 :  4.2.1 (OK)
IPython >=4.0;<6.0:  5.3.0 (OK)
pylint >=0.25     :  1.7.2 (OK)

```

**Comment by ccordoba12** [comment]
Did you mount the disk after rebooting?

**Comment by sound-fx** [comment]
Yes.  Eventually.  After quite some hours sorting through some oddities, we managed to unmount then remount the disk.  After that, the same Spyder session resumed good behavior.

**Comment by ccordoba12** [comment]
Ok, I see. What we could do is to check if the working directory can be changed to the directory of the last session, and if that's not possible use `$HOME` instead.

@sound-fx, does this sound like a good solution to you?

**Comment by sound-fx** [comment]
I should clarify that the error message from Spyder appeared a minute or two before I even noticed that the remote machine had lost power and was rebooting, and it would be hours later that we would finally succeed in un/re- mounting that disk.
I really don't know what more Spyder could have done, but I reported it because the message box offered that option.  I don't know whether the change that you propose could have made any difference for my rare case, but it sounds like it could help in other cases or at least not do any harm.
